### PR TITLE
Backend routes

### PR DESCRIPTION
**Pull request by JustinHsie**
Ok I made the backend CRUD routes! Lmk if it looks ok, it also works on the dev site.

Quick question, how does the frontend know where to send the api requests during production? 

During development, the requests are forwarded to the proxy localhost:5000 because I specified it in the client's package.json and made express listen on that port. But how does this work in production?

### CODE REVIEW

File: client/src/views/NewPlace/index.js [hunk 1]
```diff
@@ -29,6 +29,12 @@ class NewPlaceClass extends React.Component {
     this.props.getLists();
   }
 
+  componentDidUpdate(prevProps) {
```
Comment (Exegetech) [inline client/src/views/NewPlace/index.js:32]: What does this do? The logic doesn't look right to me. Why would a view requires a componentDidUpdate where it compares its props?

**Comment by Exegetech** [comment]
> Ok I made the backend CRUD routes! Lmk if it looks ok, it also works on the dev site.
> 
> Quick question, how does the frontend know where to send the api requests during production?
> 
> During development, the requests are forwarded to the proxy localhost:5000 because I specified it in the client's package.json and made express listen on that port. But how does this work in production?

in production since assets are served by the backend, it doesn't require proxy, it should work just like that.

File: client/src/views/NewPlace/index.js (hunk 1 above)
Comment (JustinHsie) [reply client/src/views/NewPlace/index.js:32]: When I make a new place, I need to call createPlace action and then addPlaceToList action. But I need the placeId to addPlaceToList, and I only get the new placeId asynchronously after I call createPlace.

So I first call createPlace, then after it's done it updates props with the new placeId (which is previously null), then when the placeId is available call addPlaceToList.

Originally, the createPlace action had two API calls - create the place, then add place to list. But then I thought it wasn't good for one action function to do two things, making it not pure I think? So I changed it to this. 
Sorry it's confusing, what's the best way to do this?

Comment (Exegetech) [reply client/src/views/NewPlace/index.js:32]: Hmm, I am not sure. I'm not really following just by a glance in the code. Let's just keep it this way.

**Comment by Exegetech** [comment]
Ok just merge it.

What's next? User?

**Comment by JustinHsie** [comment]
Ok let's do user next!

Should we save username/password or do Oauth?

**Pull request merged**